tests/unit/query_helpers/test_question_validator.py

```diff
@@ -16,18 +18,16 @@ def question_validator():
     return QuestionValidator()
 
 
-def test_invalid_response(question_validator, monkeypatch):
+@patch("ols.src.query_helpers.question_validator.LLMChain")
+@patch("ols.src.query_helpers.question_validator.LLMLoader")
+def test_invalid_response(llm_loader, llm_chain, question_validator):
     """Test how invalid responses are handled by QuestionValidator."""
     # response not in the following set should generate a ValueError
     # [INVALID,NOYAML]
     # [VALID,NOYAML]
     # [VALID,YAML]
 
-    ml = mock_llm_chain({"text": "default"})
-    monkeypatch.setattr(ols.src.query_helpers.question_validator, "LLMChain", ml)
-    monkeypatch.setattr(
-        ols.src.query_helpers.question_validator, "LLMLoader", mock_llm_loader()
-    )
+    llm_chain = mock_llm_chain({"text": "default"})
```

Comment:
this is somehow replacing the llm_chain variable in question_validator.py?  i'm not understand how this works.

but it certainly looks cleaner